dxr/plugins/rust/menu.py

```diff
@@ -38,446 +37,137 @@ def truncate_value(value, typ=""):
-    target_name = 'trait method'
-
-
-class JumpToDefinitionMenuMaker(_JumpToTarget, _RustPluginAttr):
-    target_name = 'definition'
-
+    return {'html':   "Find references",
+            'title':  "Find references to this " + kind,
+            'href':   search_url(tree_config, "+" + filter_name + ":%s" % quote(qualname)),
+            'icon':   'reference'}
 
-class JumpToModuleDefinitionMenuMaker(_JumpToTarget, _RustPluginAttr):
-    target_name = 'module defintion'
 
+def std_lib_links((doc_url, src_url, dxr_url), extra_text=""):
+    # TODO: Stop storing entire URLs in ES.
+    def get_domain(url):
+        start = url.find('//') + 2
+        return url[start:url.find('/', start)]
 
-class JumpToModuleDeclarationMenuMaker(_JumpToTarget, _RustPluginAttr):
-    target_name = 'module declaration'
+    def add_link_to_menu(url, html, title):
+        if url:
+            menu.append({'html': html,
+                         'title': title,
+                         'href': url,
+                         'icon': 'jump'})
 
-
-class JumpToAliasDefinitionMenuMaker(_JumpToTarget, _RustPluginAttr):
-    target_name = 'alias defintion'
-
-
-class JumpToCrateMenuMaker(_JumpToTarget, _RustPluginAttr):
-    target_name = 'crate'
+    menu = []
+    add_link_to_menu(doc_url,
+                     'Go to docs' + extra_text,
+                     'Go to documentation for this crate on ' + get_domain(doc_url))
+    add_link_to_menu(src_url,
+                     'Go to source' + extra_text,
+                     'Go to source code for this crate on ' + get_domain(doc_url))
+    add_link_to_menu(dxr_url,
+                     'Go to DXR index' + extra_text,
+                     'Go to DXR index of this crate on ' + get_domain(doc_url))
+    return menu
 
 
-class JumpToTypeDeclarationMenuMaker(_JumpToTarget, _RustPluginAttr):
-    target_name = 'type declaration'
+def call_menu(qualname, tree):
+    return [{'html': "Find callers",
+             'title': "Find functions that call this function",
+             'href': search_url(tree, "+callers:%s" % quote(qualname)),
+             'icon': 'method'},
+            {'html': "Find callees",  # TODO: Probably useless. Remove.
+             'title': "Find functions that are called by this function",
+             'href': search_url(tree, "+called-by:%s" % quote(qualname)),
+             'icon': 'method'}]
 
 
-class JumpToVariableDeclarationMenuMaker(_JumpToTarget, _RustPluginAttr):
-    target_name = 'variable declaration'
+def function_menu_generic(datum, tree_config):
+    """Return menu makers shared by function def/decls and function refs."""
+    menus = call_menu(datum['qualname'], tree_config)
+    menus.append(find_references_menu(tree_config, datum['qualname'], "function-ref", "function"))
+    return menus
 
 
-class JumpToFunctionDeclarationMenuMaker(_JumpToTarget, _RustPluginAttr):
-    target_name = 'function declaration'
+def jump_to_target_menu(tree_config, path, row, target_name=None):
```

Comment:
It would be very bad for target_name to be omitted, right? So let's not give it a default value.